gprproject/gprbuild.py

```diff
@@ -91,13 +91,13 @@ def __init__(
         if self.original_target:
             gprconfig_cmd.append(f"--target={self.original_target}")
         gprconfig_output = self.capture(gprconfig_cmd)
-        self.target = re.findall(r" 1 normalized_target:(\S*)", gprconfig_output)[0]
```

Comment:
I think we will need both cases: If a default compiler is available, the line that begins with "*" is indeed the choice to make. Otherwise, we need to take the first line, so the one that begins with "1".

Reply:
My understanding is that we should only proceed if a default compiler is available.

Reply:
> My understanding is that we should only proceed if a default compiler is available.

Before answering this question, I’ve thought a bit more about this PR in general. `gprconfig` is always called without the `--config` parameter in `gprbuild.py`. As far as I understand, this `--config` parameter is required to have a default compiler. So, in our case, we should not have any default compiler, so no line beginning with `*`. Could you describe how you managed to detect an error on your side? Thanks!

Reply:
No, `gprconfig` is **always** called with `--config` parameter in `gprbuild.py`, please read `gprbuild.py:90` on `master`
In our case, the COTS is built and installed in 3 flavor, using the default Ada runtime (static and relocatable) , and using a specific Ada runtime (static only). In both cases the current `re.findall` call would return empty array